frontend/src/app/main/ui/workspace/comments.cljs

```diff
@@ -0,0 +1,535 @@
+   [app.config :as cfg]
+   [app.main.data.workspace :as dw]
+   [app.main.data.workspace.comments :as dwcm]
+   [app.main.data.workspace.common :as dwc]
+   [app.main.refs :as refs]
+   [app.main.store :as st]
+   [app.main.streams :as ms]
+   [app.main.ui.components.dropdown :refer [dropdown]]
+   [app.main.data.modal :as modal]
+   [app.main.ui.hooks :as hooks]
+   [app.main.ui.icons :as i]
+   [app.main.ui.keyboard :as kbd]
+   [app.main.ui.workspace.colorpicker]
+   [app.main.ui.workspace.context-menu :refer [context-menu]]
+   [app.util.time :as dt]
+   [app.util.dom :as dom]
+   [app.util.object :as obj]
+   [beicon.core :as rx]
+   [app.util.i18n :as i18n :refer [t tr]]
+   [cuerdas.core :as str]
+   [okulary.core :as l]
+   [rumext.alpha :as mf]))
+
+(declare group-threads-by-page)
+(declare apply-filters)
+
+(mf/defc resizing-textarea
+  {::mf/wrap-props false}
+  [props]
+  (let [value       (obj/get props "value" "")
+        on-focus    (obj/get props "on-focus")
+        on-blur     (obj/get props "on-blur")
+        placeholder (obj/get props "placeholder")
+        on-change   (obj/get props "on-change")
+
+        on-esc      (obj/get props "on-esc")
+
+        ref         (mf/use-ref)
+        ;; state       (mf/use-state value)
+
+        on-key-down
+        (mf/use-callback
+         (fn [event]
+           (when (and (kbd/esc? event)
+                      (fn? on-esc))
+             (on-esc event))))
+
+        on-change*
+        (mf/use-callback
+         (mf/deps on-change)
+         (fn [event]
+           (let [content (dom/get-target-val event)]
+             (on-change content))))]
+
+
+    (mf/use-layout-effect
+     nil
+     (fn []
+       (let [node (mf/ref-val ref)]
+         (set! (.-height (.-style node)) "0")
+         (set! (.-height (.-style node)) (str (+ 2 (.-scrollHeight node)) "px")))))
+
+    [:textarea
+     {:ref ref
+      :on-key-down on-key-down
+      :on-focus on-focus
+      :on-blur on-blur
+      :value value
+      :placeholder placeholder
+      :on-change on-change*}]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Workspace
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(mf/defc reply-form
+  [{:keys [thread] :as props}]
+  (let [show-buttons? (mf/use-state false)
+        content       (mf/use-state "")
+
+        on-focus
+        (mf/use-callback
+         #(reset! show-buttons? true))
+
+        on-blur
+        (mf/use-callback
+         #(reset! show-buttons? false))
+
+        on-change
+        (mf/use-callback
+         #(reset! content %))
+
+        on-cancel
+        (mf/use-callback
+         #(do (reset! content "")
+              (reset! show-buttons? false)))
+
+        on-submit
+        (mf/use-callback
+         (mf/deps thread @content)
+         (fn []
+           (st/emit! (dwcm/add-comment thread @content))
+           (on-cancel)))]
+
+    [:div.reply-form
+     [:& resizing-textarea {:value @content
```

Comment:
It would be nice to submit the comment with Ctrl + Intro, due that plain Intro inserts a new line.